tests/suites/os/tests/config-json/index.js

```diff
@@ -85,46 +98,21 @@ module.exports = {
 
 				const context = this.context.get();
 
-				test.comment(`Waiting for os-config-json service to be inactive...`);
-				return context.systemd.waitForServiceState(
-						'os-config-json.service',
-						'inactive',
-						context.link
-				).then(() => {
-					test.comment(`Setting ntpServers to ${ntpServer()} in config.json...`);
-					return context.worker.executeCommandInHostOS(
-						[
-							`tmp=$(mktemp)`,
-							`&&`, `cat`, `/mnt/boot/config.json`,
-							`|`, `jq`, `'.ntpServers="${ntpServer()}"'`, `>`, `$tmp`,
-							`&&`, `mv`, `"$tmp"`, `/mnt/boot/config.json`,
-						].join(' '),
-						this.link);
-				}).then(() => {
-					test.comment(`Waiting for balena-ntp-config service to be active...`);
-					return context.systemd.waitForServiceState(
-						'balena-ntp-config.service',
-						'active',
-						context.link,
-					)
-				}).then(() => {
+				return setConfig(test, context, 'ntpServers', ntpServer()) 
+				.then(() => {
 					return test.resolves(
-						context.worker.executeCommandInHostOS(
-							`chronyc sources | grep ${ntpServer('.*')}`,
-							context.link,
-						),
-						'Device should show one record with our ntp server',
+						this.utils.waitUntil(async () => {
+							return context.worker.executeCommandInHostOS(
+								`chronyc sources | grep -q ${ntpServer('.*')} ; echo $?`,
+								context.link,
+							).then((exitCode) => {
+								return Promise.resolve(exitCode === '0');
+							});
+						}, false, 60, 500),
```

Comment:
And two other commonly used functions I can think about:
* Execute a command and grep for a match
* Dump and empty logs, execute a command and look for a match in the logs.

Abstracting common action would simplify reading and writing tests.